File: apps/s-2-admin/src/api/apiKey/ApiKey.ts
import { User } from "../user/User";
import { AppModel } from "../appModel/AppModel";

export type ApiKey = {
  id: string;
  note: string | null;
  createdAt: Date;
  expiresAt: Date | null;
  lastUsedAt: Date | null;
  hashedKey: string;
  user?: User | null;
  appField?: AppModel | null;
};

File: apps/cactus-demo-dotnet/src/APIs/ApiKey/Dtos/ApiKey.cs
namespace CactusDemoDotnet.APIs.Dtos;

public class ApiKey
{
    public string Id { get; set; }

    public string? Note { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime? ExpiresAt { get; set; }

    public DateTime? LastUsedAt { get; set; }

    public string HashedKey { get; set; }

    public int? User { get; set; }

    public string? AppField { get; set; }
}
